### PR TITLE
Add docs to soroban-auth and align function names with docs

### PR DESCRIPTION
**Pull request by leighmcculloch**
### What
Add docs to soroban-auth. Also rename check_auth to verify. Also rename get_identifier to identifier.

### Why
We need more docs for soroban-auth that at the least help the developer understand the concepts. The docs in this PR are a starting point.

We still need examples in these docs, but this is a starting point.

I renamed `check_auth` to `verify` because I noticed `verify`ing is how the existing docs described the function. If that's the concept we use to describe and teach about what the function does, we should name it that.

I renamed `get_identifier` to `identifier` because in Rust it is idiomatic to name getters of data that do not otherwise perform any other action as simply the name that is being got. There is some details about this https://github.com/rust-lang/rfcs/blob/master/text/0344-conventions-galore.md#gettersetter-apis. It's not a hard and fast rule, but the soroban-sdk follows this pattern where it can, and this should too.

For any renames deprecated messages for the old functions are added.

### CODE REVIEW

File: soroban-auth/src/lib.rs [hunk 1]
```diff
@@ -1,3 +1,11 @@
+//! Soroban auth provides basic authentication capabilities to Soroban
+//! contracts.
+//!
+//! See [`verify`] for how to use.
+//!
+//! **The utilities in this create provide no replay protection. Contracts must
```
Comment (sisuresh) [inline soroban-auth/src/lib.rs:6]: Nit: create -> crate

File: soroban-auth/src/lib.rs [hunk 2]
```diff
@@ -1,3 +1,11 @@
...
+//! provide their own mechanism suitable replay prevention that prevents
```
Comment (sisuresh) [inline soroban-auth/src/lib.rs:7]: suitable *for* replay prevention

File: soroban-auth/src/lib.rs [hunk 3]
```diff
@@ -61,20 +79,50 @@ fn check_account_auth(env: &Env, auth: &AccountSignatures, function: Symbol, arg
...
+///   An ed25519 key can sign [`SignaturePayload`] and include that signature in
+///   the `sig` field.
+///
+/// - Account Signatures
+///
+///   An account's signers can sign [`SignaturePayload`] and include those
+///   signatures in the `sig` field.
+///
+/// **This function provides no replay protection. Contracts must provide their
+/// own mechanism suitable replay prevention that prevents contract invocations
```
Comment (sisuresh) [inline soroban-auth/src/lib.rs:107]: suitable *for* replay prevention

File: soroban-auth/src/public_types.rs [hunk 4]
```diff
@@ -40,15 +58,39 @@ pub enum Identifier {
...
+///
+/// - `contract`
+///
+///   The contract ID for the function being invoked.
+///
+/// - `function`
+///
+///   The symbol for the function being invoked.
+///
+/// Applications using the signing the signature payload must take care to only
```
Comment (sisuresh) [inline soroban-auth/src/public_types.rs:78]: I think `the signing` needs to be removed.

File: tests/token/src/lib.rs [hunk 5]
```diff
@@ -0,0 +1,118 @@
+#![no_std]
```
Comment (sisuresh) [inline tests/token/src/lib.rs:1]: This contract doesn't have replay prevention. Was that intentional?

Comment (leighmcculloch) [reply tests/token/src/lib.rs:1]: Woops, that wasn't meant to be part of this PR. But yeah it doesn't have replay, its just a hack I'm using to play with apis.

**Pull request merged**